pan-domain-auth-example/app/controllers/AdminController.scala

```diff
@@ -1,19 +1,24 @@
 package controllers
 
-import play.api.mvc.{ Action, Controller}
+import javax.inject.Inject
+
+import play.api.Configuration
+import play.api.mvc.{Action, Controller}
 import play.api.Play.configuration
 import play.api.Play.current
+import play.api.libs.ws.WSClient
 
 
-object AdminController extends Controller with ExampleAuthActions {
+class AdminController @Inject() (override val config: Configuration, override val wsClient: WSClient)
```

Comment:
Yep - I finally bit the bullet and worked out how to do compile time DI - the flexible-restorer project is a beautiful example of this. However, it adds two more classes and I didn't think it was worth the hassle in this case. Having said that it would be pretty easy.